dom-renderers/DOMRenderer.js

```diff
@@ -612,14 +612,16 @@ DOMRenderer.prototype.setAttribute = function setAttribute(name, value) {
  *
  * @method
  *
- * @param {String} content Content to be set as `innerHTML`
+ * @param {String|HTMLElement} content A string to be set as `innerHTML` or an
+ *                                     instance of HTMLElement to be set as the
+ *                                     child element.
  *
  * @return {undefined} undefined
  */
 DOMRenderer.prototype.setContent = function setContent(content) {
     this._assertTargetLoaded();
 
-    if (this._target.formElement) {
+    if (this._target.isInputElement || this._target.isTextAreaElement) {
```

Comment:
Not checking here for `<select>` elements because select elements can have inner html, so it's possible that someone might want to use a template/view library like React to dynamically change that inner html.